aggregator/list.go

```diff
@@ -110,7 +112,7 @@ func newMetricList(resolution time.Duration, opts Options) *metricList {
 		resolution:    resolution,
 		flushInterval: flushInterval,
 		aggregations:  list.New(),
-		encoder:       msgpack.NewAggregatedEncoder(encoderPool.Get()),
+		encoder:       msgpack.NewAggregatedEncoder(encoder),
```

Comment:
can we do the encoder.Reset() in encoderPool.Get()? so you don't have to call this every time.